lib/src/core/engine.dart

```diff
@@ -657,7 +666,23 @@ class Engine extends Disposable with EventsEmittable<EngineEvent> {
       fastConnectOptions: fastConnectOptions,
     );
 
+    bool publisherConnected =
+        publisher?.pc.connectionState?.isConnected() ?? false;
+
+    if (publisher != null && !publisherConnected) {
+      logger.warning('restartConnection: Waiting for publisher to connect...');
+      await events.waitFor<EnginePublisherPeerStateUpdatedEvent>(
```

Comment:
I think there's a bit of a race condition here. if peer connection is connected between the previous check and here. we would missed it and have thrown a timeout?

very small window I know. I think similar issues may exist elsewhere too.

Reply:
It should not be here, the pc has been disposed of before the re-connection.
``` dart
await publisher?.dispose();
publisher = null;
......
if (publisher != null && !publisherConnected) {
```

Reply:
yes.. but I think `connect` will recreate it and establish the connection?

anyways, here it seems like the connection would most definitely take longer than it takes this function to execute the next line.. but more wondering the general issue of using `events.waitFor` while the conditions have become true before that call is made.